alternative-routes/map.js

```diff
@@ -0,0 +1,201 @@
+import mapboxgl from 'mapbox-gl';
+import { displayRouteData } from './index';
+import { getDurationString } from '../utils';
+
+// eslint-disable-next-line no-undef
+mapboxgl.accessToken = process.env.MAPBOX_TOKEN;
+
+const map = new mapboxgl.Map({
+  container: 'map',
+  style: 'mapbox://styles/chargetrip/ck98fwwp159v71ip7xhs8bwts',
+  zoom: 5,
+  center: [8.1320104, 52.3758916],
+});
+
+// Display the charge time on a hover
+const popup = new mapboxgl.Popup({
+  closeButton: false,
+  closeOnClick: false,
+});
+
+/**
+ * Draw route polyline and show charging stations on the map.
+ *
+ * @param coordinates {array} Array of coordinates
+ * @param legs {array} route legs (stops) - each leg represents either a charging station, or via point or final point
+ */
+export const drawRoutes = routes => {
+  if (map.loaded()) {
+    routes.forEach((route, index) => drawPolyline(route, index, index === 0 ? '#0078FF' : '#9CA7B2'));
+    map.moveLayer(`0`);
+    showLegs(routes[0].data.legs);
+  } else {
+    map.on('load', () => {
+      routes.forEach((route, index) => drawPolyline(route, index, index === 0 ? '#0078FF' : '#9CA7B2'));
+      map.moveLayer(`0`);
+      showLegs(routes[0].data.legs);
+    });
+  }
+
+  for (let i = 0; i < routes.length; i++) {
+    map.on('mouseenter', `${i}`, e => {
+      map.getCanvas().style.cursor = 'pointer';
+      const coordinates = e.lngLat;
+      const description =
+        `<b>${getDurationString(routes[i].data.duration ?? 0)}</b>` + '</br>' + `${routes[i].data.charges ?? 0} stops`;
+
+      while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+        coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+      }
+
+      popup
+        .setLngLat(coordinates)
+        .setHTML(description)
+        .addTo(map);
+    });
+
+    map.on('click', `${i}`, () => {
+      changeRoute(routes, i);
+    });
+
+    map.on('mouseleave', `${i}`, () => {
+      map.getCanvas().style.cursor = '';
+      popup.remove();
+    });
+  }
+
+  const routeOptions = document.querySelectorAll('input[type=radio][name="routes"]');
+  routeOptions.forEach((route, index) => route.addEventListener('change', () => changeRoute(routes, index)));
```

Comment:
looks like we can just pass here `routes[index]` and then no need to pass the entire routes array?  And I would also switch position around meaning first pass id and then the object. Usually it's done in this order :)

Reply:
I am passing through all routes because we need to change the color to grey for the routes that are not selected.